### PR TITLE
Logql: Fix grammar for `String`

### PR DESCRIPTION
**Pull request by ivanahuckova**
Currently our grammar has incorrectly matched string in 3 cases:
- If the string has single quotes `'`. This is not valid in LogQL.
- If the string with quotes `"` didn't have closing `"` (more info: https://discuss.codemirror.net/t/trying-to-understand-lezer-grammar-for-string/2264)
- If the string with back-tick \` didn't have closing \` (more info: https://discuss.codemirror.net/t/trying-to-understand-lezer-grammar-for-string/2264)

We are fixing this in this PR and adding tests.

### CODE REVIEW

File: test/expression.txt [hunk 1]
```diff
@@ -87,6 +87,22 @@ LogQL(Expr(LogExpr(Selector(Matchers(Matcher(Identifier,Eq,String))),PipelineExp
...
+LogQL(Expr(LogExpr(Selector(Matchers(Matcher(Identifier, Eq, String))), PipelineExpr(PipelineStage(LineFilters(LineFilter(Filter(PipeExact), ⚠, ⚠(Identifier))))))))
+
+# Log query with pipe exact filter and unclosed back-ticks
+
+{job="loki"} |= `line
+
+==>
+
+LogQL(Expr(LogExpr(Selector(Matchers(Matcher(Identifier, Eq, String))), PipelineExpr(PipelineStage(LineFilters(LineFilter(Filter(PipeExact), ⚠, ⚠(Identifier))))))))
+
```
Comment (svennergr) [inline test/expression.txt:105]:
```suggestion
# Log query with pipe exact filter expression and escaped quotes

{job="loki"} |= "hello\"quotes\""

==>

LogQL(Expr(LogExpr(Selector(Matchers(Matcher(Identifier,Eq,String))),PipelineExpr(PipelineStage(LineFilters(LineFilter(Filter(PipeExact),String)))))))

# Log query with pipe exact line filter expression and back-ticks and quotes

{job="loki"} |= `hello"quotes"`

==>

LogQL(Expr(LogExpr(Selector(Matchers(Matcher(Identifier,Eq,String))),PipelineExpr(PipelineStage(LineFilters(LineFilter(Filter(PipeExact),String)))))))

```

Do you think it is worth adding those?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@grafana/lezer-logql",
-  "version": "0.0.15",
+  "version": "0.0.16",
```
Comment (svennergr) [inline package.json:3]: The package-lock.json should also be updated.

**Review by svennergr — COMMENTED** [review]
I think the package-lock.json is mandatory. The rest is looking good.

File: package.json (hunk 2 above)
Comment (ivanahuckova) [reply package.json:3]: Yes good catch, I've committed the fix.

File: test/expression.txt (hunk 1 above)
Comment (ivanahuckova) [reply test/expression.txt:120]: Yes, absolutely. More tests, the better.

**Review by svennergr — APPROVED** [review]
LGTM!

**Pull request merged**